src/sentry/organizations/services/organization/methods/list_organizations.py

```diff
@@ -0,0 +1,136 @@
+        org_results = []
+        for org in sorted(queryset, key=lambda x: x.name):
+            # O(N) query
+            org_results.append(
+                {"organization": serialize(org), "singleOwner": org.has_single_owner()}
+            )
+
+        return Response(org_results)
+
+    elif not (actor_is_active_superuser and show == "all"):
+        queryset = queryset.filter(
+            id__in=OrganizationMember.objects.filter(user_id=actor_user_id).values("organization")
+        )
+        if actor_organization_id is not None and queryset.count() > 1:
+            # If a token is limited to one organization, this endpoint should only return that one organization
+            queryset = queryset.filter(id=actor_organization_id)
+
+    if query:
+        tokens = tokenize_query(query)
+        for key, value in tokens.items():
+            if key == "query":
+                query_value = " ".join(value)
+                user_ids = {
+                    u.id
+                    for u in user_service.get_many_by_email(emails=[query_value], is_verified=False)
+                }
+                queryset = queryset.filter(
+                    Q(name__icontains=query_value)
+                    | Q(slug__icontains=query_value)
+                    | Q(member_set__user_id__in=user_ids)
+                )
+            elif key == "slug":
+                queryset = queryset.filter(in_iexact("slug", value))
+            elif key == "email":
+                user_ids = {
+                    u.id for u in user_service.get_many_by_email(emails=value, is_verified=False)
+                }
+                queryset = queryset.filter(Q(member_set__user_id__in=user_ids))
+            elif key == "platform":
+                queryset = queryset.filter(
+                    project__in=ProjectPlatform.objects.filter(platform__in=value).values(
+                        "project_id"
+                    )
+                )
+            elif key == "id":
+                queryset = queryset.filter(id__in=value)
+            elif key == "status":
+                try:
+                    queryset = queryset.filter(
+                        status__in=[OrganizationStatus[v.upper()] for v in value]
+                    )
+                except KeyError:
+                    queryset = queryset.none()
+            elif key == "member_id":
+                queryset = queryset.filter(
+                    id__in=OrganizationMember.objects.filter(id__in=value).values("organization")
+                )
+            else:
+                queryset = queryset.none()
+
+    paginator_cls: type[PaginatorLike]
+    if sort_by == "members":
+        queryset = queryset.annotate(member_count=Count("member_set"))
+        order_by = "-member_count"
+        paginator_cls = OffsetPaginator
+    elif sort_by == "projects":
+        queryset = queryset.annotate(project_count=Count("project"))
```

Comment:
confirm: are projects on control? i think they should be

Reply:
Projects are not available in Control Silo: https://github.com/getsentry/sentry/blob/59c3578b85f3625ead0ef9dbc17c773f988b3ec2/src/sentry/models/project.py#L228

Similarly, organizations are not fully available in Control Silo either: https://github.com/getsentry/sentry/blob/59bd9f3feceb138928da7506fae2360cbedd1b2d/src/sentry/models/organization.py#L144

We do have a thin replicated version of organizations in Control though ([OrganizationMapping](https://github.com/getsentry/sentry/blob/52173d47570b492f80742b3329602d865b45ea2d/src/sentry/models/organizationmapping.py#L19-L20))

I don't have the full context on that particular decision, though @markstory could probably weigh in on this.